### PR TITLE
[receiver/dockerstats] For V2 implementation, create sensible enabled defaults 

### PR DESCRIPTION
**Pull request by jamesmoessis**
**Description:** 

This is not a breaking change unless you are using the non-default V2 scraper which is behind a feature gate. 

Currently almost every metric is enabled by default, which was done to demonstrate feature parity with the original V1 scraper which enables almost everything by default. 

This changes so most metrics are not enabled by default, and instead a sensible set of defaults are enabled. There are many very detailed memory and block IO metric that 99% of users will never need, and all it does it consume needless MTS.

The change brings the default set of metrics from 63 to 14. 

The new list of enabled metrics by default is below. I made it similar to the defaults of the signalfx agent. I am open to suggestions on changes here. 

- `container.cpu.usage.system`
- `container.cpu.usage.total`
- `container.cpu.usage.kernelmode`
- `container.cpu.usage.usermode`
- `container.cpu.percent`
- `container.memory.usage.limit`
- `container.memory.usage.total`
- `container.memory.percent`
- `container.memory.total_cache`
- `container.blockio.io_service_bytes_recursive`
- `container.network.io.usage.rx_bytes`
- `container.network.io.usage.tx_bytes`
- `container.network.io.usage.rx_dropped`
- `container.network.io.usage.tx_dropped`

**Link to tracking Issue:** 

https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9794

**Testing:** 

All unit tests still test 100% of the metrics. 

**Documentation:**

Documentation automatically updated by `go generate` and `mdatagen`.

### CODE REVIEW

**Comment by jamesmoessis** [comment]
FYI @MovieStoreGuy @rmfitzpatrick

**Review by MovieStoreGuy — APPROVED** [review]

File: unreleased/dockerstats-defaults.yaml
Comment (TylerHelmuth) [inline unreleased/dockerstats-defaults.yaml:1]: This feels like a breaking change. Since this component is alpha that totally ok, but I think the changelog should reflect that.  Users should be able to read the changelog and quickly see this could affect their deployments.

Comment (jamesmoessis) [reply unreleased/dockerstats-defaults.yaml:1]: Yeah, I was on the fence about this. It's because it's only breaking for anyone that has explicitly enabled a featuregate that to my knowledge no one is using. I am happy to change it to a breaking change though, will do that now.

Comment (jamesmoessis) [reply unreleased/dockerstats-defaults.yaml]: Alright I've updated and included subtext about the context in which it is actually breaking.

**Review by TylerHelmuth — APPROVED** [review]
LGTM

**Review by rmfitzpatrick — APPROVED** [review]

**Comment by jamesmoessis** [comment]
@TylerHelmuth this has your approval, codeowner, plus one more. Please merge whenever possible 🙂

**Review by codeboten — APPROVED** [review]

**Pull request merged**
